service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsControllerMidEventTest.java

```diff
@@ -0,0 +1,92 @@
+@OverrideAutoConfiguration(enabled = true)
+class UploadC2DocumentsControllerMidEventTest {
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String ERROR_MESSAGE = "A document must be uploaded";
+    private static final String DOCUMENT_URL = "http://localhost/documents/85d97996-22a5-40d7-882e-3a382c8ae1b4";
+    private static final String BINARY_URL = "http://localhost/documents/85d97996-22a5-40d7-882e-3a382c8ae1b4/binary";
+    private static final String FILENAME = "file.pdf";
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldReturnAnErrorWhenDocumentIsNotUploaded() throws Exception {
+        CallbackRequest request = createCallbackRequestWithTempC2Bundle(null,null,null);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldNotReturnAnErrorWhenDocumentIsUploaded() throws Exception {
+        CallbackRequest request = createCallbackRequestWithTempC2Bundle(DOCUMENT_URL,BINARY_URL,FILENAME);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
+    private CallbackRequest createCallbackRequestWithTempC2Bundle(String documentUrl,
+                                                                  String binaryUrl,
+                                                                  String filename) {
+        return CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of(
+                    "temporaryC2Document",
+                    documentOrEmptyMap(documentUrl, binaryUrl, filename)))
+                .build())
+            .build();
+    }
+
+    private ImmutableMap documentOrEmptyMap(String documentUrl, String binaryUrl, String filename) {
+        if (isNull(documentUrl)) {
+            return ImmutableMap.of();
+        } else {
+            return ImmutableMap.of(
+                "document", ImmutableMap.of(
+                    "document_url", documentUrl,
+                    "document_binary_url", binaryUrl,
+                    "document_filename", filename));
+        }
+    }
+
+    private AboutToStartOrSubmitCallbackResponse makeRequest(CallbackRequest request) throws Exception {
+        MvcResult response = mockMvc
+            .perform(post("/callback/upload-c2/mid-event")
+                .header("authorization", AUTH_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(mapper.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        return mapper.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+    }
+}
```

Comment:
could we combine the two endpoints we're testing into one file? we can refactor makeRequest to be like:
`    private MvcResult performResponseCallBack(CallbackRequest request, String endpoint) throws Exception {
        return mockMvc
            .perform(post("/callback/upload-c2/" + endpoint)
                .header("authorization", AUTH_TOKEN)
                .header("user-id", USER_ID)
                .contentType(MediaType.APPLICATION_JSON)
                .content(mapper.writeValueAsString(request)))
            .andExpect(status().isOk())
            .andReturn();
    }`
and then pass in the endpoint we're testing. in this way we could have all our integration tests in one file - what do you think? if no, then, rename existing UploadC2DocumentsControllerTest so that it reflects which endpoint it tests

Reply:
if we merge into one file we would change the test names accordingly to reflect the endpoint it tests:
i.e midEventShouldNotReturnAnErrorWhenDocumentIsUploaded()

Reply:
will do, thanks :)